### DISCUSSION TITLE
[Elastic] How to gracefully notify of a worker node failure (on a Ray cluster) ?

### DISCUSSION THREAD

**Discussion by ASDen** [horovod/horovod#3386] [Q&A]
Hi,

I understand elastic Horovod support two modes of resumming operation upon a node failure: graceful (HostsUpdatedInterrupt) and non-graceful (HorovodInternalError)
My question is, on a public cloud (e.g. GCP), using Horovod on Ray, if I can catch locally (on the node) the notification of a node eviction/preemption, how can I raise this to the rest of the world before a failure happens ? tried running `ray stop` on that node and many other things, but always a failure happens first, then a non-graceful rollback+ host blacklist

The only way I could test the graceful behavior was using the `TestDiscovery` class provided in the examples

Thanks!

**Comment by ASDen** [comment]
just an update, even if I called `hvd.shutdown()` before exiting it still gives 
> ERROR:horovod.ray.elastic:10.128.0.85[1]:The actor died unexpectedly before finishing this task.

and resets state

**Comment by ashahab** [comment]
How did you use TestDiscovery class?

**Reply by ASDen** [reply]
Hi @ashahab 
I used the example provided here: https://github.com/horovod/horovod/blob/master/examples/ray/basic_ray_elastic.py
great if you have input in this issue!
